sdk/storage/Azure.Storage.DataMovement/src/CommitChunkHandler.cs

```diff
@@ -73,70 +74,73 @@ public struct Behaviors
             _bytesTransferred = blockSize;
 
             _blockSize = blockSize;
+            _transferType = transferType;
+            if (_transferType == TransferType.Sequential)
+            {
+                _commitBlockHandler += GetQueueBlockEvent;
+            }
+            _commitBlockHandler += GetCommitBlockEvent;
+        }
 
-            if (transferType == TransferType.Sequential)
+        public void Dispose()
+        {
+            CleanUp();
+        }
+
+        public void CleanUp()
+        {
+            if (_transferType == TransferType.Sequential)
             {
-                AddQueueBlockEvent();
+                _commitBlockHandler -= GetQueueBlockEvent;
             }
-            AddCommitBlockEvent();
+            _commitBlockHandler -= GetCommitBlockEvent;
         }
 
-        public void AddCommitBlockEvent()
+        private async Task GetCommitBlockEvent(StageChunkEventArgs args)
```

Comment:
Sorry to bring up naming again (feel free to ignore) but with this approach maybe it's better to just call this `CommitBlockEvent` since it **is** the event? Like what you did below with `JobPartEvent`.

This would apply to all delegate functions in this review.